nd4j/src/main/scala/nexus/impl/nd4j/FloatTensor.scala

```diff
@@ -0,0 +1,77 @@
+package nexus.impl.nd4j
+
+import nexus.algebra.typelevel._
+import org.nd4j.linalg.api.ndarray.INDArray
+import org.nd4j.linalg.factory.Nd4j
+import shapeless._
+import shapeless.ops.nat._
+
+
+
+/**
+  * A tensor stored off-heap,
+  * and whose operations run either on the CPU or the GPU depening on Nd4j backend.
+  * @author Andrey Romanov
+  * @since 0.1.0
+  */
+trait FloatTensor[A] extends UntypedFloatTensor { self =>
+
+  protected def slice0[N <: Nat, T]
+  (axis: N, i: Int)
+  (implicit t: RemoveAt.Aux[A, N, T], nn: ToInt[N]): FloatTensor[T] =
+    sliceUntyped(nn(), i).typeWith[T]
+
+
+  def sliceAlong[X, N <: Nat, T]
+  (axis: X, i: Int)
+  (implicit n: IndexOf.Aux[A, X, N], t: RemoveAt.Aux[A, N, T], nn: ToInt[N]) = slice0(n(), i)
+
+  def along[X, N <: Nat, T]
+  (axis: X)
+  (implicit n: IndexOf.Aux[A, X, N], t: RemoveAt.Aux[A, N, T], nn: ToInt[N]): Seq[FloatTensor[T]] =
+    (0 until shape(nn())) map { i => slice0(n(), i) }
+
+  def expandDim[X, N <: Nat, T]
+  (axis: X, n: N)
+  (implicit d: InsertAt.Aux[A, N, X, T], nn: ToInt[N]) = ???
+
+
+  def asSeq: Seq[Float] = ???
+
+}
+
+object FloatTensor {
+  private def convertShape(shape: Array[Int]) = shape match {
+    case Array() => Array(1, 1)
+    case Array(d) => Array(d, 1)
+    case s => s
+  }
+
+  def scalar(value: Float): FloatTensor[Unit] =
+    new Tensor(Nd4j.scalar(value))
+
+  def scalar(value: Double): FloatTensor[Unit] =
+    new Tensor(Nd4j.scalar(value))
+
+  def fill[A](value: => Float, axes: A, shape: Array[Int]): FloatTensor[A] =
```

Comment:
`value` is of type `=> Float`, meaning the actual value could change for every `value` call. This is designed to do things like `fill(Gaussian.sample)`.

Reply:
Sorry but this is still wrong -- `Nd4j.ones().mul(value)` creates a tensor with all values in the tensor all being the same. What I mean for `fill(Gaussian.sample)` could results in a tensor where each element is a distinct sample from the Gaussian sample.

One way to do this is something like `Nd4j.fromArray(Array.fill(SIZE_OF_TENSOR)(value))`

Reply:
But this is exactly what I did 
https://github.com/ctongfei/nexus/pull/14/commits/5eeb03c3893bcf8d7e775222be1143fb2901814b#diff-d693ad230f56980247854c2e7e43e67dR86

Reply:
Ahh sorry I must have misread. Sorry for that.